### PR TITLE
Fix MSVC warnings when compiling the binaryen target

### PR DESCRIPTION
**Pull request by dcodeIO**
Not super important, but this fixes the warnings I've seen when compiling the `binaryen` target in Visual Studio, except

`wasm-interpreter.obj : warning LNK4221: This object file does not define any previously undefined public symbols, so it will not be used by any link operation that consumes this library`

### CODE REVIEW

File: src/ir/abstract.h [hunk 1]
```diff
@@ -43,16 +43,10 @@ enum Op {
 inline UnaryOp getUnary(Type type, Op op) {
   switch (type) {
     case i32: {
-      switch (op) {
```
Comment (kripken) [inline src/ir/abstract.h]: these are sort of for future use ;) but may as well remove them until that future happens, if it ever does...

File: src/passes/OptimizeInstructions.cpp [hunk 2]
```diff
@@ -826,7 +826,7 @@ struct OptimizeInstructions : public WalkerPass<PostWalker<OptimizeInstructions,
         } else if (auto* ext = Properties::getSignExtValue(binary)) {
           // if sign extending the exact bit size we store, we can skip the extension
           // if extending something bigger, then we just alter bits we don't save anyhow
-          if (Properties::getSignExtBits(binary) >= store->bytes * 8) {
+          if (Properties::getSignExtBits(binary) >= uint32_t(store->bytes * 8)) {
```
Comment (kripken) [inline src/passes/OptimizeInstructions.cpp]: hmm, what was the warning here? (both are unsigned already, so it wasn't that I don't think?)

The cast should be to `Index` (what the function on the left returns, and which represents the proper index size in wasm modules - 32 bit for now).

File: src/passes/Print.cpp [hunk 3]
```diff
@@ -52,7 +52,7 @@ struct PrintSExpression : public Visitor<PrintSExpression> {
 
   PrintSExpression(std::ostream& o) : o(o) {
     setMinify(false);
-    if (!full) full = isFullForced();
+    if (!full) full = isFullForced() != 0;
```
Comment (kripken) [inline src/passes/Print.cpp]: how about making `isFullForced` return a bool?

File: src/passes/OptimizeInstructions.cpp (hunk 2 above)
Comment (dcodeIO) [reply src/passes/OptimizeInstructions.cpp]: The warning was `C4018: ">=": signed/unsigned mismatch`, presumably due to integer promotion on the rhs

File: src/ir/abstract.h (hunk 1 above)
Comment (dcodeIO) [reply src/ir/abstract.h]: thought so :) the warning here was `C4065: switch statement contains 'default' but no 'case' labels`

**Comment by dcodeIO** [comment]
Appears CI failed for other reasons

**Pull request merged**